### PR TITLE
SEO Optimization

### PR DESCRIPTION
**Pull request by solverat**
Allow better category and product getter overrides.

### CODE REVIEW

File: src/CoreShop/Bundle/FrontendBundle/Controller/ProductController.php [hunk 1]
```diff
@@ -54,4 +54,13 @@ public function detailAction(Request $request)
...
         ]);
     }
+
+    /**
+     * @param Request $request
+     * @return DataObject
+     */
+    private function getProductByRequest(Request $request)
+    {
+        return $product = DataObject::getById(2);
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/FrontendBundle/Controller/ProductController.php]: ?

File: src/CoreShop/Bundle/FrontendBundle/Controller/CategoryController.php [hunk 2]
```diff
@@ -87,18 +96,19 @@ public function indexAction(Request $request)
 
         $perPage = $request->get('perPage', $defaultPerPage);
 
-        $category = $this->getRepository()->find($request->get('category'));
-
-        if (!$category instanceof CategoryInterface) {
-            return $this->redirectToRoute('coreshop_index');
+        $categoryList = $this->getRepository()->findBy([$this->repositoryIdentifier => $request->get($this->requestIdentifier)], null, 1);
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/FrontendBundle/Controller/CategoryController.php]: not quite happy with that, maybe we could add a findUniqueBy function to the pimcore class?

Comment (solverat) [reply src/CoreShop/Bundle/FrontendBundle/Controller/CategoryController.php]: done. findOneBy was already implemented.

**Comment by dpfaffenbauer** [comment]
cheers

**Pull request merged**